projects.js

```diff
@@ -0,0 +1,178 @@
+    main.className = 'main-modal';
+    const modal = document.createElement('div');
+    document.body.style.overflow = 'hidden';
+    Workitem.append(main);
+    main.append(modal);
+    modal.className = 'modal';
+    const cancelicondesktop = new Image();
+    cancelicondesktop.src = 'images/Disabled.png';
+    cancelicondesktop.className = 'cancel-icon-desktop';
+    modal.appendChild(cancelicondesktop);
+    const cancelicon = new Image();
+    cancelicon.src = 'images/cancel.png';
+    cancelicon.className = 'cancel-icon';
+    modal.appendChild(cancelicon);
+    cancelicon.addEventListener('click', () => {
+      main.remove();
+      btndiv.appendChild(btn);
+      detaildiv.appendChild(ul);
+      detaildiv.appendChild(btndiv);
+      document.body.style.overflow = 'visible';
+    });
+    cancelicondesktop.addEventListener('click', () => {
+      main.remove();
+      btndiv.appendChild(btn);
+      detaildiv.appendChild(ul);
+      detaildiv.appendChild(btndiv);
+      document.body.style.overflow = 'visible';
+    });
+
+    const imagediv = document.createElement('div');
+    imagediv.className = 'image-div';
+    const image = new Image();
+    image.src = work.image;
+    imagediv.appendChild(image);
+    Workitem.appendChild(imagediv);
+    modal.appendChild(image);
+    image.className = 'img-project';
+    const h2 = document.createElement('h2');
+    h2.className = 'title-project';
+    h2.textContent = work.title;
+    modal.appendChild(h2);
+    const Uldetail = document.createElement('ul');
+    Uldetail.className = 'tech-stack-detail';
+    const array = work.technologies;
+    array.forEach((value) => {
+      const Lidetail = document.createElement('li');
+      Lidetail.textContent = value;
+      Uldetail.appendChild(Lidetail);
+    });
+    modal.appendChild(Uldetail);
+    const p = document.createElement('p');
+    p.className = 'description';
+    p.textContent = work.description1;
+    const p2 = document.createElement('p');
+    p2.textContent = work.description2;
+    p2.className = 'description';
+    modal.appendChild(p);
+    modal.appendChild(p2);
+    const Btndivinner = document.createElement('div');
+    Btndivinner.className = 'btndiv';
+    const source = document.createElement('a');
+    const demo = document.createElement('a');
+    demo.className = 'btn';
+    source.className = 'btn';
+    demo.textContent = 'See live';
+    source.textContent = 'See source';
+    const iconlive = new Image(20, 20);
+    iconlive.className = 'project-icons';
+    iconlive.src = 'images/live-demo.png';
+    const iconsource = new Image();
+    iconsource.className = 'project-icons';
+    iconsource.src = 'images/source.png';
+    demo.appendChild(iconlive);
+    source.appendChild(iconsource);
+    imgdiv.appendChild(img);
+    demo.setAttribute('href', work.live_version);
+    source.setAttribute('href', work.source);
+    Btndivinner.appendChild(demo);
+    Btndivinner.appendChild(source);
```

Comment:
- [ ] [OPTIONAL] This dynamic display of project items code is too large and not easy to follow. I suggest that you break down this code into smaller functions and then invoke each function when it should. See code below  👇🏾 

```javascript
function createWorkItem(work) {
  const Workitem = document.createElement('div');
  Workitem.className = 'work-item';
  const imgdiv = createImageDiv(work);
  Workitem.appendChild(imgdiv);

  const detaildiv = createDetailsDiv(work);
  Workitem.append(detaildiv);
  return Workitem;
}

function createImageDiv(work) {
  const imgdiv = document.createElement('div');
  imgdiv.className = 'snapshot';
  const img = new Image();
  img.src = work.image;
  img.className = 'image';
  imgdiv.appendChild(img);
  return imgdiv;
}

function createDetailsDiv(work) {
  const detaildiv = document.createElement('div');
  detaildiv.className = 'details';
  const heading = createHeading(work.title);
  detaildiv.appendChild(heading);

  const ul = createUL(work.technologies);
  detaildiv.appendChild(ul);

  const btndiv = createButtonDiv(work);
  detaildiv.appendChild(btndiv);

  return detaildiv;
}

function createHeading(title) {
  const heading = document.createElement('h2');
  heading.textContent = title;
  return heading;
}

function createUL(technologies) {
  const ul = document.createElement('ul');
  ul.className = 'tech-stack';
  technologies.forEach((value) => {
    const li = document.createElement('li');
    li.textContent = value;
    ul.appendChild(li);
  });
  return ul;
}

function createButtonDiv(work) {
  const btndiv = document.createElement('div');
  btndiv.className = 'project';
  const btn = createButton(work);
  btndiv.appendChild(btn);
  return btndiv;
}

function createButton(work) {
  const btn = document.createElement('button');
  btn.textContent = 'See Project';
  btn.addEventListener('click', () => {
    showModal(work);
  });
  return btn;
}

function showModal(work) {
  const main = createMain(work);
  const modal = createModal(work);
  const cancelicondesktop = createCancelIconDesktop();
  const cancelicon = createCancelIcon();
  cancelicon.addEventListener(
    'click',
    closeProject(main, btndiv, ul)
  );
  cancelicondesktop.addEventListener(
    'click',
    closeProject(main, btndiv, ul)
  );

  const imagediv = createImageDiv(work);
  modal.appendChild(imagediv);
  const h2 = createH2(work.title);
  modal.appendChild(h2);
  const Uldetail = createULDetail(work.technologies);
  modal.appendChild(Uldetail);
  const p = createP(work.description1);
  const p2 = createP(work.description2);
  modal.appendChild(p);
  modal.appendChild(p2);
  const Btndivinner = createButtonDivInner(work);
  modal.appendChild(Btndivinner);

  Workitem.appendChild(imagediv);
  main.append(modal);
  btndiv.appendChild(cancelicondesktop);
  btndiv.appendChild(cancelicon);
  document.body.style.overflow = 'hidden';
}

function createMain() {
  const main = document.createElement('div');
  main.className = 'main-modal';
  Workitem.append(main);
  return main;
}

function createModal() {
  const modal = document.createElement('div');
  modal.className = 'modal';
  return modal;
}

function createCancelIconDesktop() {
  const cancelicondesktop = new Image();
  cancelicondesktop.src = 'images/Disabled.png';
  cancelicondesktop.className = 'cancel-icon-desktop';
  return cancelicondesktop;
}

function createCancelIcon() {
  const cancelicon = new Image();
  cancelicon.src = 'images/cancel.png';
  cancelicon.className = 'cancel-icon';
  return cancelicon;
}

function closeProject(main, btndiv, ul) {
  return () => {
    main.remove();
    btndiv.appendChild(createButton(work));
    detaildiv.appendChild(ul);
    detaildiv.appendChild(btndiv);
    document.body.style.overflow = 'visible';
  };
}

function createH2(title) {
  const h2 = document.createElement('h2');
  h2.className = 'title-project';
  h2.textContent = title;
  return h2;
}

function createULDetail(technologies) {
  const Uldetail = document.createElement('ul');
  Uldetail.className = 'tech-stack-detail';
  technologies.forEach((value) => {
    const Lidetail = document.createElement('li');
    Lidetail.textContent = value;
    Uldetail.appendChild(Lidetail);
  });
  return Uldetail;
}

function createP(description) {
  const p = document.createElement('p');
  p.className = 'description';
  p.textContent = description;
  return p;
}

```

This is not all, but it is enough to give you an example of what it should be.